### PR TITLE
Adding requirements.txt to manage global developer python dependencies

### PR DESCRIPTION
**Pull request by jbobo**
Added a requirements.txt for developers to install python dependencies (linter will return import errors if all dependencies are not installed locally). I need create tasks to:
- add instructions to readme.md on using requirements.txt
- add requirements.txt for each container directory
- add a script to bin for updating all requirements.txt files

Run `pip3 install -r requirements.txt` in the root project directory to install development dependencies for the project.

### CODE REVIEW

**Comment by devsatishm** [comment]
@jbobo Use `develop` branch instead of `master` as your base for the PRs.

**Comment by jbobo** [comment]
@devsatishm I thought we were still deciding how to move forward on that issue?

@mtn217 and I fixed his issue by running `sudo apt-get install pkg-config`

**Comment by adamgering** [comment]
> @jbobo Use `develop` branch instead of `master` as your base for the PRs.

We've abandoned the develop branch until a sync process has been determined.

**Comment by adamgering** [comment]
Is this supposed to be a part of PR #174?

**Comment by jbobo** [comment]
@adamgering this is more for developers to install required dependencies so that they can run the linter as required in CONTRIBUTING.md, while PR #174 is for deployment. Should I make these one single PR?

**Comment by adamgering** [comment]
Only requirement to run docker and docker tests is pyaml, and maybe we'll want to add black.
Why are the others here?

File: requirements.txt [hunk 1]
```diff
@@ -0,0 +1,15 @@
+itsdangerous==0.24
```
[the remark this answers is not in the document]
Comment (adamgering) [reply requirements.txt]: No, don't make these one PR.

[the remark this answers is not in the document]
Comment (adamgering) [reply requirements.txt]: /requirements.txt is used by Travis and should not be also used for development dependencies.

[the remark this answers is not in the document]
Comment (adamgering) [reply requirements.txt]: I'm reasonable certain this list of dependencies exceeds the minimum required to run build and lint. Please verify.

**Comment by adamgering** [comment]
I can't install either one of these. Anyone else able or unable to install these dependencies? (Python 3.7 / OSX)
sawtooth_sdk==1.0.5
sawtooth_signing==1.0.5

**Comment by adamgering** [comment]
I suggest renaming this requirements_dev.txt so we don't conflict with the travis default.

**Comment by jbobo** [comment]
> Only requirement to run docker and docker tests is pyaml, and maybe we'll want to add black.
> Why are the others here?

We can't run the linter locally without all of these dependencies.

**Comment by jbobo** [comment]
@adamgering should we edit CONTRIBUTORS.md then? 

> Please ensure that lint passes by running ‘./bin/run_lint’. The command should produce no output if there are no lint errors.

/requirements.txt was added under the assumption that we were going to keep this as a requirement. if not should I remove requirements.txt and change CONTRIBUTORS.md to: 

> Please ensure that lint passes by running:
> ‘docker exec -it rbac-shell bash bin/run_lint’ 
> The command should produce no output if there are no lint errors.

**Comment by adamgering** [comment]
> @adamgering should we edit CONTRIBUTORS.md then?
> 
> > Please ensure that lint passes by running ‘./bin/run_lint’. The command should produce no output if there are no lint errors.
> 
> /requirements.txt was added under the assumption that we were going to keep this as a requirement. if not should I remove requirements.txt and change CONTRIBUTORS.md to:
> 
> > Please ensure that lint passes by running:
> > `docker exec -it rbac-shell bash bin/run_lint`
> > The command should produce no output if there are no lint errors.

Sure

**Comment by jbobo** [comment]
@adamgering I made the requested changes.

**Comment by jbobo** [comment]
@adamgering I noticed that the rbac-shell container exits with code 0 shortly after running `docker-compose up` how should we start and run the shell container to run the linter?

**Pull request merged**